src/test/scala/org/apache/spark/sql/delta/MergeIntoSuiteBase.scala

```diff
@@ -1122,41 +1129,55 @@ abstract class MergeIntoSuiteBase
       name: String)(
       mergeOn: String,
       mergeClauses: MergeClause*)(
-      errorStrs: Seq[String]): Unit = {
+      errorStrs: Seq[String],
+      notErrorStrs: Seq[String] = Nil): Unit = {
     test(s"extended syntax analysis errors - $name") {
-      withKeyValueData(source = Seq.empty, target = Seq.empty) { case (sourceName, targetName) =>
-        val e = intercept[AnalysisException] {
+      withKeyValueData(
+        source = Seq.empty,
+        target = Seq.empty,
+        sourceKeyValueNames = ("key", "newValue"),
+        targetKeyValueNames = ("key", "oldValue")
```

Comment:
tbh this `oldValue` is really confusing. Can this just be `value`? I was really confused to see `oldValue = newValue`.